specification/under_development/Instruction_Set.md

```diff
@@ -0,0 +1,52 @@
+# Instruction Set
+
+A backend that supports quantum computations is composed of multiple processing units that include one or more quantum chips as well as classic computing resources.
+A quantum program is a combination of instructions that execute on a quantum processor, 
+instructions that execute on an adjacent classical processor, and data transfer between processors.
+
+The QIR profile defines which instructions are used/supported for classical computations and data transfer, while the Quantum Instruction Set (QIS) defines which instructions are supported by the quantum processor. The profile and QIS can be defined/chosen mostly independently, with the caveat that a profile specification may contain restrictions that need to be satisfied for an QIS to be compatible (compliant) with that profile.
+To target a program to a specific backend hence requires selecting both a profiles and QIS that is supported by that backend, see also [Compilation and Targeting](Compilation_And_Targeting.md). 
+
+## Runtime Functions
+
+Instructions that execute on a classical processor or serve data transfer are expressed as LLVM instructions, including calls to `__quantum__rt__*` functions.
+These functions are forward declared in the IR and defined by the executing runtime. Which LLVM instructions and runtime functions are used/supported is captured in the QIR profile specification. The complete list of all runtime functions that are needed for executing QIR programs independent on which profile they have been compiled for is given in the [Library_Reference.md](). Most backends do not support the full set of runtime functions but merely a subset that is sufficient for the supported QIR profile(s). The [Base Profile](profiles/Base_Profile.md) defines the minimal set of runtime functions that need to be supported for by a backend to permit computations that make use of qubits and measurements of qubits.Additional functions may be needed to support the use of data types beyond those for qubit and measurement result values.
```

Comment:
"These functions [...] are defined by the executing runtime." Probably I'm misunderstanding something here, but doesn't the compiler (which is generating the QIR to send to the runtime) need to know the definitions? Or does this just mean that the runtime decides how to implement them at a low level?

Reply:
By defining I meant as in a definition in the code. So yes, the intent was to say that the spec formalizes what they are, and the runtime provides the implementation. I'll see that I clarify that.